### PR TITLE
Switch to truncate to allocate storage for images

### PR DESCRIPTION
**Pull request by Vaelatern**
Works in low memory situations
#50 XOR #51

### CODE REVIEW

**Comment by Gottox** [comment]
There's a portable way for truncating:
```
dd if=/dev/null bs=1M seek="$((ROOTFS_SIZE+ROOTFS_FREESIZE))" of="$BUILDDIR/tmp/LiveOS/ext3fs.img"
```
(Keep in mind, that ```M``` suffixes are not portable)

**Comment by chneukirchen** [comment]
That argument is invalid since the script is linux-dependent already, and even busybox has truncate.

The benefit of truncate is that it doesn't actually write zeroes to the disk, which is much faster.

File: mklive.sh.in [hunk 1]
```diff
@@ -200,7 +200,7 @@ generate_grub_efi_boot() {
     modprobe -q loop || :
 
     # Create EFI vfat image.
-    dd if=/dev/zero of=$GRUB_DIR/efiboot.img bs=1024 count=16384  >/dev/null 2>&1
+    truncate -s "$((1024*16384))M" $GRUB_DIR/efiboot.img >/dev/null 2>&1
```
Comment (Vaelatern) [inline mklive.sh.in]: @chneukirchen this computation?

Comment (chneukirchen) [inline mklive.sh.in]: Yes, that's 16 Terabyte.

Comment (Vaelatern) [inline mklive.sh.in]: Should it be `16384M`?

Comment (pullmoll) [inline mklive.sh.in]: It has been 16KiB before.

Comment (Vaelatern) [inline mklive.sh.in]: Fixed to `16M`

**Pull request merged**